state/protocol/convert.go

```diff
@@ -66,14 +67,26 @@ func ToEpochCommit(epoch Epoch) (*flow.EpochCommit, error) {
 	if err != nil {
 		return nil, fmt.Errorf("could not get epoch clustering: %w", err)
 	}
-	qcs := make([]*flow.QuorumCertificate, 0, len(clustering))
+	qcs := make([]*flow.QuorumCertificateWithSignerIDs, 0, len(clustering))
 	for i := range clustering {
 		cluster, err := epoch.Cluster(uint(i))
 		if err != nil {
 			return nil, fmt.Errorf("could not get epoch cluster (index=%d): %w", i, err)
 		}
-		qcs = append(qcs, cluster.RootQC())
+		qc := cluster.RootQC()
+		// TODO: double check cluster.Members returns canonical order
+		signerIDs, err := packer.DecodeSignerIdentifiersFromIndices(cluster.Members().NodeIDs(), qc.SignerIndices)
```

Comment:
Good catch. There is some inconsistency here we may want to resolve:

* `Cluster.Members` returns identities in the order [specified by the cluster assignment list](https://github.com/onflow/flow-go/blob/c59b5e4de1b59766d352209dbc856068c03f9225/model/flow/cluster.go#L59-L64) (respects order of `EpochSetup` event).
* `Epoch.InitialIdentities` returns identities in the [canonical order](https://github.com/onflow/flow-go/blob/bd910749460f48d77823af422bf8baeb798b2b13/model/convert/service_event.go#L317)
* `Snapshot.Identities` returns identities in the [canonical order](https://github.com/onflow/flow-go/blob/bd910749460f48d77823af422bf8baeb798b2b13/state/protocol/badger/snapshot.go#L227)

Reply:
The Cluster.Members is created from the AssignmentList. So [I fixed from the source](https://github.com/onflow/flow-go/pull/2140/files#r828751067) by sorting the assignment list so that when cluster list is built off the assignment, it's already sorted.